MonumentAddons.cs

```diff
@@ -315,16 +315,46 @@ private void OnTelekinesisStopped(BasePlayer player, BaseEntity moveEntity, Base
             SingleEntityAdapter adapter;
             SingleEntityController controller;
 
-            if (!_entityTracker.IsMonumentEntity(moveEntity, out adapter, out controller))
-                return;
+            int adapterCount;
+            string profileName;
 
-            if (!adapter.TrySaveAndApplyChanges())
+            var spawnPointAdapter = GetSpawnPointAdapter(moveEntity);
+
+            if (spawnPointAdapter != null)
+            {
+                if (!spawnPointAdapter.Monument.IsInBounds(moveEntity.transform.position))
+                    return;
+
+                var localPosition = spawnPointAdapter.Monument.InverseTransformPoint(moveEntity.transform.position);
+
+                spawnPointAdapter.SpawnPointData.Position = localPosition;
+                spawnPointAdapter.SpawnPointData.RotationAngles = (Quaternion.Inverse(spawnPointAdapter.Monument.Rotation) * moveEntity.transform.rotation).eulerAngles;
+                spawnPointAdapter.SpawnPointData.SnapToGround = IsOnTerrain(moveEntity.transform.position);
+                spawnPointAdapter.SpawnPointData.SnapToTerrain = IsOnTerrain(moveEntity.transform.position);
```

Comment:
`SnapToTerrain` determines whether the spawn point itself snaps to the terrain. `SnapToGround` is specific to spawn points and determines whether the entities spawned by the spawn point will snap to the ground (not necessarily terrain). Updating the former makes sense, since that's what we're doing when a `SingleEntityAdapter` is moved, but I don't think we should be updating `SnapToGround` here.

Reply:
Sorry for delay, had a vacation, new job... cant believe its month already. Anyway I removed `SnapToGround` updating, but when this option is set (it is set by default) entity will be moved to ground after telekinesis stopped, while spawn point stays above